CMakeLists.txt

```diff
@@ -64,16 +70,24 @@ if(AVIF_LOCAL_ZLIBPNG)
     # Put the value of ZLIB_INCLUDE_DIR in the cache. This works around cmake behavior that has been updated by
     # cmake policy CMP0102 in cmake 3.17. Remove the CACHE workaround when we require cmake 3.17 or later. See
     # https://gitlab.kitware.com/cmake/cmake/-/issues/21343.
-    set(ZLIB_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ext/zlib" CACHE PATH "zlib include dir")
+    set(ZLIB_INCLUDE_DIR
+        "${CMAKE_CURRENT_SOURCE_DIR}/ext/zlib"
+        CACHE PATH "zlib include dir")
     include_directories("${CMAKE_CURRENT_BINARY_DIR}/ext/zlib")
     set(CMAKE_DEBUG_POSTFIX "")
 
     # This is the only way I could avoid libpng going crazy if it found awk.exe, seems benign otherwise
     set(PREV_ANDROID ${ANDROID})
     set(ANDROID TRUE)
-    set(PNG_BUILD_ZLIB "${CMAKE_CURRENT_SOURCE_DIR}/ext/zlib" CACHE STRING "" FORCE)
-    set(PNG_SHARED OFF CACHE BOOL "")
-    set(PNG_TESTS OFF CACHE BOOL "")
+    set(PNG_BUILD_ZLIB
+        "${CMAKE_CURRENT_SOURCE_DIR}/ext/zlib"
+        CACHE STRING "" FORCE)
+    set(PNG_SHARED
+        OFF
+        CACHE BOOL "")
+    set(PNG_TESTS
+        OFF
+        CACHE BOOL "")
```

Comment:
I think it's a good idea to format the cmake files with a tool, but the style file results in many short lines such as the three `set()` calls here (lines 82-90) and the `project()` call at lines 9-12.

Is it possible to avoid this problem?

I believe libaom also formats its cmake files with a tool, but its cmake files contains `set()` calls like the following:

```
      set(${config} "${${config}} ${c_flag}" CACHE STRING "" FORCE)
```
and
```
    set(${name} ${value} CACHE STRING "${helpstring}")
```

This gives me hope that it is possible.

Reply:
I believe that's max_subargs_per_line, it's set to 10 in libaom (the default is 3).

Reply:
Indeed there is the config file [`.cmake-format.py`](https://aomedia.googlesource.com/aom/+/refs/heads/main/.cmake-format.py) in libaom. Should I reuse the entire file in libavif? Or just add `max_subargs_per_line`?

I used YAML for the config file to match `.travis.yml` and `appveyor.yml` but python is fine too.

Reply:
I suggest we aim to minimize the reformatting changes, so we should figure out the config settings for the current prevalent style in libavif. This is likely to require some experimentation. We can't use libaom's config file directly because the tab_size is different (2 vs. 4).

It seems that the config file consists of name-value pairs, so it doesn't matter whether the config file is written in YAML of Python.

Reply:
(`max_subargs_per_line` [does no longer exist](https://github.com/cheshirekow/cmake_format/blob/904c42da298f7fda932d9ac774517c20e2667fb8/cmakelang/doc/release_notes.rst#v060) in cmake-format 0.6.3. It was replaced by `max_subgroups_hwrap` and `max_pargs_hwrap`.)